### PR TITLE
Add exported claims

### PR DESCRIPTION
**Pull request by shawn-hurley**
## Related issue(s)
Part of #1219

Fixes #

### CODE REVIEW

File: pkg/admission/permissionclaims/mutating_permission_claims.go [hunk 1]
```diff
@@ -161,21 +161,16 @@ func (m *mutatingPermissionClaims) Validate(ctx context.Context, a admission.Att
 // SetKcpInformers implements the WantsExternalKcpInformerFactory interface.
 func (m *mutatingPermissionClaims) SetKcpInformers(f kcpinformers.SharedInformerFactory) {
 	m.apiBindingsHasSynced = f.Apis().V1alpha1().APIBindings().Informer().HasSynced
-	m.apiExportsHasSynced = f.Apis().V1alpha1().APIExports().Informer().HasSynced
```
Comment (ncdc) [inline pkg/admission/permissionclaims/mutating_permission_claims.go:164]: remove the struct fields too

File: pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go [hunk 2]
```diff
@@ -218,7 +216,7 @@ func (c *controller) process(ctx context.Context, key string) error {
 			return fmt.Errorf("failed to create patch for apibinding %s|%s: %w", clusterName, name, err)
 		}
 
-		logger.V(2).Info("patching APIBinding", "patch", string(patchBytes))
+		klog.V(2).InfoS("Patching apibinding", "cluster", clusterName, "name", name, "patch", string(patchBytes))
```
Comment (ncdc) [inline pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go:219]: undo switch to raw klog

File: pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go [hunk 3]
```diff
@@ -119,6 +119,7 @@ func (c *controller) reconcile(ctx context.Context, apiBinding *apisv1alpha1.API
 				applyErrors.Insert(s)
 			}
 			continue
+
```
Comment (ncdc) [inline pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go:122]: undo

File: test/e2e/virtual/apiexport/virtualworkspace_test.go [hunk 4]
```diff
@@ -326,7 +326,23 @@ func TestAPIExportPermissionClaims(t *testing.T) {
...
 
 	framework.Eventually(t, func() (success bool, reason string) {
-		// Get the binding and make sure that observed permission claims are all set.
+		// Get the binding and make sure that observed export permission claims are all set.
+		binding, err := kcpClusterClient.ApisV1alpha1().APIBindings().Get(logicalcluster.WithCluster(ctx, consumerWorkspace), "cowboys", metav1.GetOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+
+		for _, claim := range binding.Status.ExportPermissionClaims {
```
Comment (nrb) [inline test/e2e/virtual/apiexport/virtualworkspace_test.go:335]: Would there ever be a case where `Status.ExportPermissionClaims` and `Status.AppliedPermissionClaims` would not be the same?

File: pkg/apis/apis/v1alpha1/types_apiexport.go [hunk 5]
```diff
@@ -206,7 +206,7 @@ func (p PermissionClaim) String() string {
 func (p PermissionClaim) Equal(claim PermissionClaim) bool {
 	return p.Group == claim.Group &&
 		p.Resource == claim.Resource &&
-		p.IdentityHash == p.IdentityHash
+		p.IdentityHash == claim.IdentityHash
```
Comment (ncdc) [inline pkg/apis/apis/v1alpha1/types_apiexport.go:209]: Nice catch!

File: pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go [hunk 6]
```diff
@@ -187,12 +187,10 @@ func (c *controller) process(ctx context.Context, key string) error {
 
 	old := obj
 	obj = obj.DeepCopy()
-
```
Comment (ncdc) [inline pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go:190]: I personally prefer the whitespace, but 🤷

File: pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go [hunk 7]
```diff
@@ -187,12 +187,10 @@ func (c *controller) process(ctx context.Context, key string) error {
...
 	var errs []error
 	if err := c.reconcile(ctx, obj); err != nil {
 		errs = append(errs, err)
 	}
-
```
Comment (ncdc) [inline pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go:195]: I personally prefer the whitespace, but 🤷

File: test/e2e/virtual/apiexport/virtualworkspace_test.go (hunk 4 above)
Comment (ncdc) [reply test/e2e/virtual/apiexport/virtualworkspace_test.go:335]: Potentially, yes. It can take time to label things (this has to finish before AppliedPermissionClaims is updated)

File: pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go [hunk 8]
```diff
@@ -218,7 +215,7 @@ func (c *controller) process(ctx context.Context, key string) error {
 			return fmt.Errorf("failed to create patch for apibinding %s|%s: %w", clusterName, name, err)
 		}
 
-		logger.V(2).Info("patching APIBinding", "patch", string(patchBytes))
+		logger.V(2).Info("Patching apibinding", "cluster", clusterName, "name", name, "patch", string(patchBytes))
```
Comment (ncdc) [inline pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go:218]: Can undo this - the logger already has the APIBinding in its values

File: test/e2e/virtual/apiexport/virtualworkspace_test.go [hunk 9]
```diff
@@ -326,7 +326,23 @@ func TestAPIExportPermissionClaims(t *testing.T) {
...
+			if claim.IdentityHash == identityHash {
```
Comment (ncdc) [inline test/e2e/virtual/apiexport/virtualworkspace_test.go:336]: I'm pretty sure we want to verify that all the claims in this list match those in the APIExport, right?

Comment (nrb) [reply test/e2e/virtual/apiexport/virtualworkspace_test.go:336]: That's what I thought; I was mostly following the pattern in the AppliedPermissionClaims, though I think that one's incorrect too. Or at least, the comment doesn't match the code.

Comment (ncdc) [reply test/e2e/virtual/apiexport/virtualworkspace_test.go:336]: Let's at least make that change here

File: pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go [hunk 10]
```diff
@@ -88,7 +88,6 @@ func NewController(
 	})
 
 	return c, nil
-
```
Comment (ncdc) [inline pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go:91]: If you undo this change and the case change on L220/221, then there are no changes to this file in this PR 😄

**Review by ncdc — APPROVED** [review]
